File: HelpSoftware/Help/Help.EF/Entitys/UserFavedArticles.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Help.EF
{
  public class UserFavedArticles : IHelpEntity
  {
    [Key]
    public virtual int ID_UserFavedArticles { get; set; }

    public virtual User FavedUser { get; set; }

    public virtual Article FavedArticle { get; set; }

    public virtual DateTime FavedTime { get; set; }

    [Timestamp]
    public byte[ ] TimeStamp { get; set; }

    public string GetEntityName()
    {
      return "UserFavedArticles";
    }

    public int GetID()
    {
      return ID_UserFavedArticles;
    }

    public bool GetWriteHistoryEntry()
    {
      return false;
    }
  }
}

File: HelpSoftware/Help/Help.Web/Web/src/app/EF/UserFavedArticles.ts
import { User, Article } from "./EF";

export class UserFavedArticles {
  ID_UserFavedArticles: number;
  FavedUser: User;
  FavedArticle: Article;
  FavedTime: Date;

  deserialize(input): UserFavedArticles {
    this.ID_UserFavedArticles = input.ID_UserFavedArticles;
    this.FavedUser = new User().deserialize(input.FavedUser);
    this.FavedArticle = new Article().deserialize(input.Article);
    this.FavedTime = input.FavedTime;
    return this;
  }
}
